src/player.cpp

```diff
@@ -5636,6 +5639,23 @@ void player::on_effect_int_change( const efftype_id &eid, int intensity, body_pa
     morale->on_effect_int_change( eid, intensity, bp );
 }
 
+const targeting_data &player::get_targeting_data()
+{
+    if( tdata == nullptr ) {
+        debugmsg( "Tried to get targeting data before setting it" );
+        tdata.reset( new targeting_data() );
+        tdata->relevant = nullptr;
+        cancel_activity();
+    }
+
+    return *tdata;
+}
+
+void player::set_targeting_data( const targeting_data &td )
+{
+    tdata.reset( new targeting_data( td ) );
+}
+
```

Comment:
I'm not sure how this got in my PR...

Reply:
These functions and `#include "ranged.h"` were removed in #38605 if it helps

Reply:
Thank you!